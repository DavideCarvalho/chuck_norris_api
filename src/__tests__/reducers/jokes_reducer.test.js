import jokesReducer from './../../reducers/jokes_reducer';


describe('jokes_reducer tests', () => {
  it('should bring default state', () => {
    expect(jokesReducer(undefined, {})).toMatchSnapshot();
  })
  it('should add jokes categories to state', () => {
    const { categories } = jokesReducer(undefined, {
      type:'FETCH_JOKES_CATEGORIES',
      payload: ['science']
    })
    expect(categories[0]).toBe('science')
  })
  it('should add empty array if payload is null', () => {
    const { categories } = jokesReducer(undefined, {
      type:'FETCH_JOKES_CATEGORIES',
      payload: null
    })
    expect(categories.length).toBe(0)
  })
  it('should add empty array if payload is an empty array', () => {
    const { categories } = jokesReducer(undefined, {
      type:'FETCH_JOKES_CATEGORIES',
      payload: []
    })
    expect(categories.length).toBe(0)
  })
  it('should add create an array if payload is a string', () => {
    const { categories } = jokesReducer(undefined, {
      type:'FETCH_JOKES_CATEGORIES',
      payload: 'science'
    })
    expect(categories[0]).toBe('science')
  })
})