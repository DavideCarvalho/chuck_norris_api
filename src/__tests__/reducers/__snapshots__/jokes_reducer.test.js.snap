// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jokes_reducer tests should bring default state 1`] = `
Object {
  "categories": Array [],
}
`;
